lib/transport/selenium-webdriver/method-mappings.js

```diff
@@ -356,7 +356,12 @@ module.exports = class MethodMappings {
         // Elements
         ///////////////////////////////////////////////////////////
         async locateSingleElement(element) {
-          const locator = Locator.create(element);
+          let locator;
+          if (this.transport.api.isAppiumClient()) {
+            locator = AppiumLocator.create(element);
+          } else {
+            locator = Locator.create(element);
+          }
```

Comment:
We can improve this by using Factory Pattern here.

```js
class Locator {
  create() {
    console.log('Locator create called');
  }
}

class AppiumLocator extends Locator {
  create() {
    console.log('AppiumLocator create called');
  }
}

class NightwatchLocator {
  static create() {
    if (isAppiumClient()) {
      return new AppiumLocator().create()
    } else {
      return new Locator().create()
    }
  }
}

// We can directly use this 
NightwatchLocator.create()
```

Reply:
I'm not sure about this, because the main reason for creating a separate class for `AppiumLocator` was to avoid passing a boolean flag (`isAppiumClient`) to `create()` method. /cc: @beatfactor

Reply:
Well this would be ok, but the main issue is to check isAppiumClient() inside the locator class.

Reply:
Please check [4fd9c47](https://github.com/nightwatchjs/nightwatch/pull/3620/commits/4fd9c477d410f2988da853570c80825666426150).